jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Job.java

```diff
@@ -54,12 +61,39 @@
     public String getName();
 
     /**
-     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are allowed.
+     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are
+     * allowed.
      * <p>
      * Default is 2 = medium.
      *
      * @return priority
      */
     public int getPriority();
 
+
+    /**
+     * @return All involved locations
+     */
+    public List<Location> getAllLocations();
+
+    /**
+     * @return Start location
+     */
+    public Location getStartLocation();
+
+    /**
+     * @return End location
+     */
+    public Location getEndLocation();
+
+    /**
+     * @return All activities
+     */
+    public JobActivityList getActivityList();
+
+    /**
+     * @return All operation time windows
+     */
+    public Set<TimeWindow> getOperationTimeWindows();
```

Comment:
Other classes do know that there are currently only two types of jobs (Service and Shipment). When you look at Shipment, you ll see that time windows are not all related to the Shipment but to the pickup and delivery activity. Therefore, I think it does not make sense provide an unrelated list of time windows of an arbitrary number of activities.